packages/next-auth/src/react.tsx

```diff
@@ -276,7 +276,7 @@ export async function signIn<
     return
   }
 
-  const error = new URL(data.url, baseUrl).searchParams.get("error")
+  const error = new URLSearchParams(data.url?.split("?")[1]).get("error")
```

Comment:
```suggestion
  const error = new URL(data.url).get("error")
```

`data.url` is supposed to always be there, but it's also supposed to be a valid URL, so we can drop `baseUrl`.

Reply:
Works for me thanks

Reply:
The suggested code will throw runtime `TypeError` since `data.url` is a relative url

```tsx
new URL("/login?error=CallbackRouteError").searchParams.get("error")
// Uncaught TypeError: Failed to construct 'URL': Invalid URL
```
```tsx
new URLSearchParams("/login?error=CallbackRouteError".split("?")[1]).get("error")
// "CallbackRouteError"

// or
new URL("/login?error=CallbackRouteError", window.location.origin).searchParams.get("error")
```